partner_contact_in_several_companies/models/res_partner.py

```diff
@@ -12,8 +12,7 @@ class ResPartner(models.Model):
         [('standalone', _('Standalone Contact')),
          ('attached', _('Attached to existing Contact')),
          ],
-        compute='_get_contact_type',
-        required=True, select=1, store=True,
+        required=True,
```

Comment:
Replacing `select=1` by `index=True` is the fix you need.

The `select` attribute has been renamed to `index`, and there was a bug in the warning displayed in case of using an old attribute name, which caused this infinite recursion traceback (trying to read attributes during attributes intialization).
This bug is now fixed, you should also update your v10 standard code: https://github.com/odoo/odoo/commit/bcddba4e4b3d9a9e62cafa74ca5fac4db55b1f46